src/components/data_submitter/ConsentGroupForm.js

```diff
@@ -0,0 +1,590 @@
+      maxLength: '512',
+      rows: '2',
+      required: required,
+      placeholder: placeholder
+    })
+  ]);
+};
+
+
+const dataLocationOptions = [
+  { value: 'AnVIL Workspace', label: 'AnVIL Workspace' },
+  { value: 'Terra Workspace', label: 'Terra Workspace' },
+  { value: 'TDR Location', label: 'TDR Location' },
+  { value: 'Not Determined', label: 'Not Determined' }
+];
+
+// const dataLocationOptions = ["AnVIL Workspace", "Terra Workspace", "TDR Location", "Not Determined"]
+
+export const ConsentGroupForm = (props) => {
+  const {
+    idx,
+    parentConsentGroup,
+    saveConsentGroup,
+    deleteConsentGroup,
+  } = props;
+
+  const [consentGroup, setConsentGroup] = useState({
+    ...{
+      consentGroupName: '',
+
+      // primary: one of these needs to be filled
+      generalResearchUse: undefined,
+      hmb: undefined,
+      diseaseSpecificUse: undefined, // string
+      poa: undefined,
+      otherPrimary: undefined, // string
+
+      // secondary:
+      nmds: false, // No Methods Development or validation studies
+      gso: false, // genetic studies only
+      pub: false, // publication required
+      col: false, // collaboration required
+      irb: false, // irb approval required
+      gs: undefined, // string: geographic restriction
+      mor: false, // publication moratorium
+      npu: false, // non profit only
+      otherSecondary: undefined, // string
+
+      // dataLocation is one of:
+      // "AnVIL Workspace", "Terra Workspace",
+      // "TDR Location", "Not Determined"
+      dataLocation: [],
+
+      url: ''
+    },
+    ...parentConsentGroup,
+  });
+
+  const [editMode, setEditMode] = useState(true);
+
+  useEffect(() => {
+    setConsentGroup(parentConsentGroup);
+  }, [parentConsentGroup]);
+
+  const updateField = (field, val) => {
+    const newConsentGroup = cloneDeep(consentGroup);
+    newConsentGroup[field] = val;
+
+    setConsentGroup(newConsentGroup);
+  };
+
+
+  const setPrimary = (field, val=true) => {
+    setConsentGroup({
+      ...consentGroup,
+      ...{
+        generalResearchUse: false,
+        hmb: false,
+        diseaseSpecificUse: undefined,
+        poa: false,
+        otherPrimary: undefined,
+      },
+      ...{
+        [field]: val,
+      }});
+  };
+
+  return div({
+    style: DataSubmitterStyles.consentGroupCard,
+    id: idx+'_consentGroupForm'
+  }, [
+
+    div({
+      isRendered: !editMode,
+      id: idx+'_consentGroupSummary',
+    }, [
+      h(ConsentGroupSummary, {
+        consentGroup: consentGroup,
+      }),
+    ]),
+
+    div({
+      isRendered: editMode,
+    }, [
+      // name
+      div({ className: 'form-group' }, [
+        label({
+          id: idx+'_consent_group_name_label',
+          className: 'control-label',
+          style: DataSubmitterStyles.header,
+        }, ['Consent Group Name*']),
+        div({ className: '' }, [
+          input({
+            type: 'text',
+            style: DataSubmitterStyles.textInput,
```

Comment:
Whenever we have these, I always think it's just better to use css, but c'est la vie! :P 

If you wanted to migrate these over to using the `FormField` objects, I dont know how much of these comments are valuable / useful